app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,34 @@
+class PostsController < ApplicationController
+  
+  def create
+  	params.permit!
+  	@post = current_user.posts.create params[:post]
+  	render @post
+  end
+  
+  def relevant
+  	render get_relevant_posts
```

Comment:
you are using a helper here when what you really want to use is a *service*.  Place the `get_relevant_posts` functionality in a `PostsService` under the `lib/services` directory in the app